xlators/mgmt/glusterd/src/glusterd.c

```diff
@@ -1621,11 +1621,39 @@ init(xlator_t *this)
         exit(1);
     }
 
-    snprintf(cmd_log_filename, PATH_MAX, "%s/cmd_history.log", logdir);
+    char *cmd_history_file = NULL;
+    ret = dict_get_str_sizen(this->options, "command-history-file",
+                             &cmd_history_file);
+    if (!cmd_history_file) {
+        snprintf(cmd_log_filename, PATH_MAX, "%s/cmd_history.log", logdir);
+    } else {
+        if (cmd_history_file[0] != '/') {
+            gf_msg(
+                this->name, GF_LOG_ERROR, errno, GD_MSG_FILE_OP_FAILED,
+                "Expects an absolute file for 'command-history-file', got %s",
+                cmd_history_file);
+            exit(1);
```

Comment:
Instead of stopping the process, wouldn't it be better to failback to the default filename ?

Reply:
This I am not ok to continue. When someone has given this as an option, we expect it to be success if the process continues. Considering this is glusterd process, having this volfile properly done once is better than realizing this error later. So, I still vote of `exit(1);`. This is inline with few other options here in the file.

Reply:
Given that there are other cases where we already do this, I'm ok with this if you prefer it, and fix it in another patch if necessary.

My concern is how it's expected to fix a bad configuration ? I mean, if I set a bad value for the option, when glusterd restarts it will automatically die, so how I will be able to fix the mistake ? we'll require the user to manually modify configuration files on all servers ?

Reply:
IMO, yes. Ie, if there is an error, user mostly needs to make changes on all the servers. Two reasons:

1. User would have manually made this change in the first place when this option is required for them.
2. We are not giving it as a default option in the file we install in standard path, so in any case, this will be useful for only those who are managing glusterd volume file externally.

Reply:
ok